tools/changelog.main.kts

```diff
@@ -0,0 +1,416 @@
+                - Navigation libraries `org.jetbrains.androidx.navigation:navigation-*:RELEASE_PLACEHOLDER`. Based on [Jetpack Navigation REDIRECT_PLACEHOLDER](https://developer.android.com/jetpack/androidx/releases/navigation#REDIRECT_PLACEHOLDER)
+                - Material3 Adaptive libraries `org.jetbrains.compose.material3.adaptive:adaptive*:RELEASE_PLACEHOLDER`. Based on [Jetpack Material3 Adaptive REDIRECT_PLACEHOLDER](https://developer.android.com/jetpack/androidx/releases/compose-material3-adaptive#REDIRECT_PLACEHOLDER)
+        
+                ___
+            """.trimIndent()
+        )
+
+        appendLine()
+        appendLine()
+    }
+}
+
+/**
+ * Transforms v1.6.0-beta01 to 1.6.0-beta01
+ */
+fun commitToVersion(commit: String) =
+    if (commit.startsWith("v") && commit.contains(".")) {
+        commit.removePrefix("v")
+    } else {
+        commit.removePrefix("release/")
+    }
+
+/**
+ * September 2024
+ */
+fun currentChangelogDate() = LocalDate.now().format(DateTimeFormatter.ofPattern("MMMM yyyy", Locale.ENGLISH))
+
+/**
+ * Formats:
+ * - A new approach to implementation of `platformLayers`. Now extra layers (such as Dialogs and Popups) drawing is merged into a single screen size canvas.
+ *
+ * to:
+ * - [A new approach to implementation of `platformLayers`](link). Now extra layers (such as Dialogs and Popups) drawing is merged into a single screen size canvas.
+ */
+fun ChangelogEntry.format(): String {
+    return if (link != null) {
+        val linkStartIndex = maxOf(
+            message.indexOfFirst { !it.isWhitespace() && it != '-' }.ifNegative { 0 },
+            message.endIndexOf("_(prerelease fix)_ ").ifNegative { 0 },
+            message.endIndexOf("(prerelease fix) ").ifNegative { 0 },
+        )
+        val linkLastIndex = message.indexOfAny(listOf(". ", " (")).ifNegative { message.length }
+
+        val beforeLink = message.substring(0, linkStartIndex)
+        val inLink = message.substring(linkStartIndex, linkLastIndex).removeLinks()
+        val afterLink = message.substring(linkLastIndex, message.length)
+
+        "$beforeLink[$inLink]($link)$afterLink"
+    } else {
+        message
+    }
+}
+
+fun Int.ifNegative(value: () -> Int): Int = if (this < 0) value() else this
+
+fun String.endIndexOf(value: String): Int = indexOf(value).let {
+    if (it >= 0) {
+        it + value.length
+    } else {
+        it
+    }
+}
+
+/**
+ * Converts:
+ * Message (title)[some link], message
+ *
+ * to:
+ * Message title, message
+ */
+fun String.removeLinks(): String = replace(Regex("\\[([^)]*)\\]\\([^\\]]*\\)"), "$1")
+
+/**
+ * Extract by format https://github.com/JetBrains/compose-multiplatform/blob/b32350459acceb9cca6b9e4422b7aaa051d9ae7d/.github/PULL_REQUEST_TEMPLATE.md?plain=1
```

Comment:
Should we refer `jb-main` instead `b32350459acceb9cca6b9e4422b7aaa051d9ae7d`?

Reply:
Done